src/Transactions/Builder/DelegateRegistration.php

```diff
@@ -60,6 +60,7 @@ public function sign(string $passphrase): AbstractTransaction
         $this->transaction->asset['delegate']['publicKey'] = $this->transaction->senderPublicKey;
 
         $this->transaction = $this->transaction->sign($keys);
+        $this->transaction->id = $this->transaction->getId();
```

Comment:
This whole method could be refactored to work like https://github.com/ArkEcosystem/php-crypto/blob/master/src/Transactions/Builder/Vote.php#L61.